scripts/quests/sandoria/PLD_AF2_A_Boys_Dream.lua

```diff
@@ -0,0 +1,235 @@
+                        if
+                            npcUtil.tradeHasExactly(trade, xi.items.GIANT_SHELL_BUG) and
+                            questProgress == 3
+                        then
+                            return quest:progressEvent(15) -- During Quest "A Boy's Dream" (trading bug) madame ?
+                        elseif
+                            npcUtil.tradeHasExactly(trade, xi.items.ODONTOTYRANNUS) and
+                            questProgress == 4
+                        then
+                            return quest:progressEvent(47) -- During Quest "A Boy's Dream" (trading odontotyrannus)
+                        end
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [15] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [25] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 7)
+                end,
+
+                [47] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 5)
+                end,
+            },
+
+        },
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Exoroche'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 2 and
+                        player:getMainJob() == xi.job.PLD
+                    then
+                        return quest:progressEvent(50)
+                    elseif questProgress >= 7 then
+                        return quest:progressEvent(32)
+                    else
+                        player:startEvent(76)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [32] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 8)
+                end,
+
+                [50] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+
+        [xi.zone.SELBINA] =
+        {
+            ['Zaldon'] =
+            {
+                onTrade = function(player, npc, trade)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress >= 4 and
+                        npcUtil.tradeHasExactly(trade, xi.items.ODONTOTYRANNUS)
+                    then
+                        return quest:progressEvent(85)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [85] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.KNIGHTS_BOOTS)
+                    quest:setVar(player, 'Prog', 6)
```

Comment:
The event that called this required progress >= 4, but then it's setting to 6 here? Is this jumping a prog in some way?

Reply:
yea it was optional to turn in the odon before getting the CS's so you can do the CS's after you get the knights boots but eventually you have to trade which is why there are the >= checks but i see what you mean - they will revert to lesser ones but then again they to complete -- ok changed all the ==